doc/rst/mason-packages/start/whymason.rst

```diff
@@ -0,0 +1,111 @@
+``mason new <package-name> --app``, the default module created will have a ``main()`` function
+already in place. This is what drives your application and where the execution will start.
+This ``main()`` function is unique to the mason application package type and makes it the only
+type of mason package that can be ran via ``mason run``.
+
+When compiling Chapel code, there can only be one ``main()`` function (or you will need to use the
+``--main-module`` flag to specify which one to run). This means that mason application package
+types are not meant to be used as dependencies for other mason packages and cannot be added as a
+dependency with the mason add <library-name@version> command. Rather, mason applications published
+to the mason registry can be downloaded, built, and run using the mason install <application-name>
+command (which is currently a work in progress). If you would like to create a library with functions
+to be exposed to other users, the mason library is what you are looking for (see :ref:`mason-libraries`).
+
+Some examples of when you would want to use a mason application as opposed to one of the other
+types are a benchmark that you would like others to be able to compile and execute in a reproducible
+way, an application that simulates the weather, or just about anywhere you have a cool program that
+you would like to share.
+
+.. _mason-libraries:
+
+Mason Libraries
+~~~~~~~~~~~~~~~
+A mason library is meant to be a module that contains functions that are to be made accessible
+to other users. Mason libraries do not have a ``main()`` function and cannot be run with the
+``mason run`` command, but Mason library code can be executed through examples
+(see :ref:`mason-examples`).
+
+To start writing your own mason library, use the mason new <package-name> --lib command and
+mason will create an empty default module. Inside the empty module you can define the
+functions your library will provide to its users.
+
+Some examples of when you would want to create a mason library as opposed to an application are:
+
+* You want to provide a linear algebra library for other Chapel programs
+
+* You have multiple Chapel projects that depend on a common set of functionality you wish to centralize
+  
+* You found some functionality missing from Chapel, and think others could benefit from your solution
+
+
+Mason Lightweight Projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+A mason lightweight project is simply a mason manifest file that tracks which mason packages your
+Chapel code depends on. This gives all Chapel users the ability to use mason libraries
+as dependencies, without forcing your project to use the folder structure defined for mason packages.
+
+Lightweight packages cannot use most mason commands, such as mason build or mason run, but a special mason modules command allows you to get the chpl compiler flags necessary to include the list of dependencies in your Chapel project.
```

Comment:
this one line is longer than others, not sure if that matters in the html docs, but stands out here